### PR TITLE
Sync statistics with the Firebase

### PR DESCRIPTION
**Pull request by onetruffle**
|PR checklist|Status| 
|-----|:------:| 
|Is issue created in GitHub?|[Yes](<https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi/issues/81>)|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?| Yes
|Is code analysis task executed and issues resolved in the CPP/native code?| Yes
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?| NA
|Is code commenting done properly for the changes made?| Yes
|Is correct variable naming convention followed?| Yes
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?| NA
|Are the message strings used in the application taken from strings.xml file?| NA
|Are the permissions which are not required removed from AndroidManifest.java file?| NA
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?| NA
|Are the generated files correctly configured in .gitignore?| NA
|Are the release signing configurations removed from the gradle files?| NA
|Is commented code removed from all files?| Yes
|Are updated images compressed (< 100 KB)?| NA
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?| Yes

### CODE REVIEW

File: launcher/app/src/main/java/com/maq/xprize/kitkitlauncher/hindi/MainActivity.java [hunk 1]
```diff
@@ -333,6 +335,22 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            if (!Settings.canDrawOverlays(this)) {
+                Toast.makeText(this, "Please give my app this permission!", Toast.LENGTH_SHORT).show();
+                Intent intent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:" + getPackageName()));
+                startActivityForResult(intent, OVERLAY_PERMISSION_REQ_CODE);
+            } else {
+                *//*Util.disableStatusBar(this);*//*
+            }
+        } else {
+            *//*Util.disableStatusBar(this);*//*
+        }*/
```
Comment (ankitk20) [inline launcher/app/src/main/java/com/maq/xprize/kitkitlauncher/hindi/MainActivity.java]: Remove commented code

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java [hunk 2]
```diff
@@ -149,6 +166,22 @@ public static String getAppVersion() {
         return "";
     }
 
+    public void firebase_setCurrentScreen(String screenName, String screenClass) {
+        if (screenName.equals("")) screenName = null;
+        if (screenClass.equals("")) screenClass = null;
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java]: Please correctly format it with braces

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java [hunk 3]
```diff
@@ -94,6 +100,17 @@ public static void clearLaunchString() {
         _launchString = "";
     }
 
+    public Cocos2dxGLSurfaceView onCreateView() {
+        glSurfaceView = new Cocos2dxGLSurfaceView(this);
+
+        this.hideSystemUI();
+
+        // create stencil buffer
+        glSurfaceView.setEGLConfigChooser(5, 6, 5, 0, 16, 8);
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java:107]: Add comment for the hardcoded integer parameters

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java [hunk 4]
```diff
@@ -67,9 +68,15 @@ of this software and associated documentation files (the "Software"), to deal
     public static AppActivity _activity;
     public static String _launchString;
     public static KitkitDBHandler _dbHandler;
+
+    private static FirebaseAnalytics mFirebaseAnalytics;
+
+    private static String TAG = "KitkitSchoolActivity";
+    private static final String EXPANSION_FILE_VERSION_KEY_NAME = "0";
```
Comment (ankitk20) [inline mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java:73]: Do not add empty lines without reason

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: launcher/app/build.gradle [hunk 5]
```diff
@@ -71,6 +72,7 @@ android {
 }
 
 dependencies {
+    compile 'com.google.firebase:firebase-core:16.0.4'
```
Comment (akashdabhi03) [inline launcher/app/build.gradle:75]: Did you try using the latest version 17.0.0?

File: mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java (hunk 3 above)
Comment (onetruffle) [reply mainapp/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java:107]: I did not put this line. Parts of the previous code were automatically removed and then added again when I was trying to resolve the conflicts.

**Review by akashdabhi03 — CHANGES_REQUESTED** [review]

File: launcher/app/build.gradle (hunk 5 above)
Comment (onetruffle) [reply launcher/app/build.gradle:75]: The build tool throws an error. The only way to use version 17.0.0 is to use the latest version of gradle which does not accept many deprecated methods which this application uses.

**Review by harry6396 — DISMISSED** [review]

**Review by ankitk20 — APPROVED** [review]

**Review by harry6396 — APPROVED** [review]

**Review by akashdabhi03 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**